pkg/webui/console/lib/feature-checks.js

```diff
@@ -0,0 +1,182 @@
+// Copyright © 2019 The Things Network Foundation, The Things Industries B.V.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { selectStackConfig } from '../../lib/selectors/env'
+import { selectApplicationRights } from '../store/selectors/applications'
+import { selectGatewayRights } from '../store/selectors/gateways'
+import { selectOrganizationRights } from '../store/selectors/organizations'
+import { selectUserRights } from '../store/selectors/user'
+
+const stackConfig = selectStackConfig()
+const asEnabled = stackConfig.as.enabled
+const gsEnabled = stackConfig.gs.enabled
+
+export const checkFromState = (featureCheck, state) =>
+  featureCheck.check(featureCheck.rightsSelector(state))
+
+// User
+export const mayViewApplications = {
```

Comment:
Though, not having both `RIGHT_USER_LIST_APPLICATIONS` and `RIGHT_ORGANIZATION_LIST_APPLICATION` would essentially mean that no applications can be viewed, right?  Then it might make sense to not show the Application tab in the navbar / overview page.
Should we bother about that?

Reply:
If at global scope (from auth_info) the user doesn't have either of those rights, you could indeed disable the application section completely. This could actually be useful, since a JS-only cluster does not need to be able to list Gateways and OAuth Clients.